lib/puppet/node.rb

```diff
@@ -45,10 +45,20 @@ def to_data_hash
       'environment' => environment.name.to_s,
     }
     result['classes'] = classes unless classes.empty?
-    result['parameters'] = parameters unless parameters.empty?
+    result['parameters'] = serializable_parameters unless serializable_parameters.empty?
     result
   end
 
+  def to_json
+    self.to_data_hash.to_json
+  end
```

Comment:
This can be dropped because all of our domain objects (`Catalog`, `Node`, etc) include `Puppet::Network::FormatSupport`:

```
irb(main):004:0> Puppet::Node.instance_method(:to_json)
=> #<UnboundMethod: Puppet::Node(Puppet::Network::FormatSupport)#to_json>
```

Reply:
Without this, we had failing tests because the environment parameter was being included in the JSON. Sounds like this is only coming up in tests because they're working with the Node in a different way than the rest of the code?

Reply:
...and I'm not seeing those test failures any more. No idea what was causing them before then.